From: Jerome Marchand <jmarchand@redhat.com>
Date: Tue, 15 Jul 2008 15:55:45 +0200
Subject: [block] Enhanced Partition Statistics: cciss fix
Message-id: 20080715135910.472405222@redhat.com
O-Subject: [Patch RHEL5.3 4/9] Enhanced Partition Statistics: cciss fix
Bugzilla: 224322
RH-Acked-by: Anton Arapov <aarapov@redhat.com>

bz224322

Updates the enhanced partition statistics in cciss driver.

commit: not needed on upstream

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index abd914a..93a247c 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -1325,7 +1325,8 @@ static void cciss_softirq_done(struct request *rq)
 	if (blk_fs_request(rq)) {
 		const int rw = rq_data_dir(rq);
 
-		disk_stat_add(rq->rq_disk, sectors[rw], rq->nr_sectors);
+		all_stat_add(rq->rq_disk, sectors[rw],
+			     rq->nr_sectors, rq->sector);
 	}
 
 #ifdef CCISS_DEBUG
